streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -820,6 +820,7 @@ public class StreamsConfig extends AbstractConfig {
             .define(STATE_DIR_CONFIG,
                     Type.STRING,
                     System.getProperty("java.io.tmpdir") + File.separator + "kafka-streams",
+                    System.getProperty("kafka.streams.state.dir"),
```

Comment:
I think it might be simpler to just hardcode what we want to put into the docs, ie, just pass in String `"${java.io.tmpdir}"`.

Reply:
Sure. I've made the change